client/app/scripts/components/timeline-control.js

```diff
@@ -0,0 +1,251 @@
+import React from 'react';
+import moment from 'moment';
+import Slider from 'rc-slider';
+import classNames from 'classnames';
+import { connect } from 'react-redux';
+import { debounce } from 'lodash';
+
+import PauseButton from './pause-button';
+import TopologyTimestampButton from './topology-timestamp-button';
+import {
+  websocketQueryInPast,
+  startWebsocketTransition,
+  clickResumeUpdate,
+} from '../actions/app-actions';
+
+import {
+  TIMELINE_SLIDER_UPDATE_INTERVAL,
+  TIMELINE_DEBOUNCE_INTERVAL,
+} from '../constants/timer';
+
+
+const sliderRanges = {
+  last15Minutes: {
+    label: 'Last 15 minutes',
+    getStart: () => moment().utc().subtract(15, 'minutes'),
+  },
+  last1Hour: {
+    label: 'Last 1 hour',
+    getStart: () => moment().utc().subtract(1, 'hour'),
+  },
+  last6Hours: {
+    label: 'Last 6 hours',
+    getStart: () => moment().utc().subtract(6, 'hours'),
+  },
+  last24Hours: {
+    label: 'Last 24 hours',
+    getStart: () => moment().utc().subtract(24, 'hours'),
+  },
+  last7Days: {
+    label: 'Last 7 days',
+    getStart: () => moment().utc().subtract(7, 'days'),
+  },
+  last30Days: {
+    label: 'Last 30 days',
+    getStart: () => moment().utc().subtract(30, 'days'),
+  },
+  last90Days: {
+    label: 'Last 90 days',
+    getStart: () => moment().utc().subtract(90, 'days'),
+  },
+  last1Year: {
+    label: 'Last 1 year',
+    getStart: () => moment().subtract(1, 'year'),
+  },
+  todaySoFar: {
+    label: 'Today so far',
+    getStart: () => moment().utc().startOf('day'),
+  },
+  thisWeekSoFar: {
+    label: 'This week so far',
+    getStart: () => moment().utc().startOf('week'),
+  },
+  thisMonthSoFar: {
+    label: 'This month so far',
+    getStart: () => moment().utc().startOf('month'),
+  },
+  thisYearSoFar: {
+    label: 'This year so far',
+    getStart: () => moment().utc().startOf('year'),
+  },
+};
+
+class TimelineControl extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      showSliderPanel: false,
+      millisecondsInPast: 0,
+      rangeOptionSelected: sliderRanges.last1Hour,
+    };
+
+    this.renderRangeOption = this.renderRangeOption.bind(this);
+    this.handleTimestampClick = this.handleTimestampClick.bind(this);
+    this.handleJumpToNowClick = this.handleJumpToNowClick.bind(this);
+    this.handleSliderChange = this.handleSliderChange.bind(this);
+    this.debouncedUpdateTimestamp = debounce(
+      this.updateTimestamp.bind(this), TIMELINE_DEBOUNCE_INTERVAL);
+  }
+
+  componentWillMount() {
+    // Force periodic re-renders to update the slider position as time goes by.
+    this.timer = setInterval(() => { this.forceUpdate(); }, TIMELINE_SLIDER_UPDATE_INTERVAL);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timer);
+    this.updateTimestamp();
```

Comment:
Whenever the component is removed and we have no way of controlling the time, we always want to regress to showing the current state. That currently happens when switching to the resource view, where the time travel component is disabled.

Actually `time-travel` is a much better name than `timeline`, I think I will just rename it everywhere :)